data_model/src/transaction.rs

```diff
@@ -744,6 +734,12 @@ mod http {
             self
         }
 
+        /// Set creation time of transaction
+        pub fn set_creation_time(&mut self, creation_time_ms: u64) -> &mut Self {
```

Comment:
the same

Reply:
I don't see why this method should be private, user might want to specify exact timestamp for his transaction.

Reply:
but then you have to modify the signature

Reply:
ok. Is `creation_time` even required to be part of the payload and signed?

Reply:
yes, otherwise it will be hard to distinguish two transfers

Reply:
could the `creation_time` be assigned by the node instead of the client?

Reply:
i don't this so because this will mess up transaction signature

Reply:
yes, it won't be a part of transaction signature. I'm questioning whether it makes sense to sign this piece of information? We're trusting a user that they will use a correct time, we might be trusting the peer

Reply:
@mversic if transaction has wrong time it will be just rejected either as being outdated or from future

Reply:
> @mversic if transaction has wrong time it will be just rejected either as being outdated or from future

I know this. My intention here is to remove `create_time_ms` from transaction that user submits so that we can remove `pending_transaction` endpoint all together. It's the only thing that stands in the way of MST

Reply:
I'm not sure that this will end up well. 
So currently we use time + nonce to distinguish between otherwise identical transactions and time is convenient way to do so.

If we would remove time from the transaction user have to use unique nonce every time he send transaction. 
And we back to square one where user need to get pending transactions in order to get proper nonce.

We might consider that creation of MST is happens outside iroha: that interested parties can communicate and create proper MST. 
This would allow to simplify:
1. CLI by not querying `/pending` transactions on every command
2. Queue because we can reject transactions with invalid signature check immediately we won't have partial MST which might or might not get additional signatures.

Reply:
thanks for the thoughtful response